tests/suites/test_suite_psa_crypto.function

```diff
@@ -3152,6 +3152,862 @@ exit:
+        {
+            if( (input_data->len - tag_length - part_offset ) <
+                ( uint32_t ) data_part_len )
+            {
+                part_length = ( input_data->len - tag_length - part_offset );
+            }
+            else
+            {
+                part_length = data_part_len;
+            }
+
+            status = psa_aead_update( &operation,
+                                      ( input_data->x + part_offset ),
+                                      part_length, part_data,
+                                      part_data_size, &output_part_length );
+
+            if( status != PSA_SUCCESS )
+            {
+                TEST_EQUAL( status, expected_result_arg );
+                goto exit;
+            }
+
+            if( output_data && output_part_length )
+            {
+                memcpy( ( output_data + part_offset ), part_data,
+                        output_part_length );
+            }
+
+            part_offset += part_length;
+            output_length += output_part_length;
+        }
+    }
+    else
+    {
+        status = psa_aead_update( &operation, input_data->x,
+                                  ( input_data->len - tag_length ), output_data,
+                                  output_size, &output_length );
+
+        if( status != PSA_SUCCESS )
+        {
+            TEST_EQUAL( status, expected_result_arg );
+            goto exit;
+        }
+    }
+
+    status = psa_aead_verify( &operation, final_data,
+                                 PSA_AEAD_VERIFY_OUTPUT_MAX_SIZE,
+                                 &output_part_length,
+                                 ( input_data->x + input_data->len -
+                                   tag_length ),
+                                 tag_length );
+
+    if( status != PSA_SUCCESS )
+    {
+        TEST_EQUAL( status, expected_result_arg );
+        goto exit;
+    }
+
+    if( output_data && output_part_length )
+    {
+        memcpy( ( output_data + output_length ), final_data,
+                output_part_length );
+    }
+
+    output_length += output_part_length;
+
+    if( expected_result != PSA_ERROR_INVALID_ARGUMENT )
+    {
+        /* For all currently defined algorithms, PSA_AEAD_DECRYPT_OUTPUT_SIZE
+         * should be exact. */
+        TEST_EQUAL( output_length,
+                    PSA_AEAD_DECRYPT_OUTPUT_SIZE( key_type, alg,
+                                                  input_data->len ) );
+        TEST_ASSERT( output_length <=
+                     PSA_AEAD_DECRYPT_OUTPUT_MAX_SIZE( input_data->len ) );
+    }
+
+    if( expected_result == PSA_SUCCESS )
+        ASSERT_COMPARE( expected_data->x, expected_data->len,
+                        output_data, output_length );
+
+exit:
+    psa_destroy_key( key );
+    psa_aead_abort( &operation );
+    mbedtls_free( output_data );
+    mbedtls_free( part_data );
+    mbedtls_free( final_data );
+    PSA_DONE( );
+}
+/* END_CASE */
+
```

Comment:
Missing: negative tests. Lots of them. For calling functions when the operation is in the wrong state, or with bad arguments.

Reply:
Hopefully I have addressed this.